src/main/scala/io/treeverse/clients/LakeFSContext.scala

```diff
@@ -45,4 +48,31 @@ object LakeFSContext {
       classOf[WithIdentifier[Catalog.Entry]],
     )
   }
+
+  def newDF(
+      spark: SparkSession,
+      repoName: String,
+      commitID: String,
+  ): DataFrame = {
+    val rdd = newRDD(spark.sparkContext, repoName, commitID).map { pair =>
+      val key = pair._1
+      val entry = pair._2.message
+      Row(
+        new String(key),
+        entry.getAddress,
+        entry.getETag,
+        new java.sql.Timestamp(
+          TimeUnit.SECONDS.toMillis(entry.getLastModified.getSeconds),
+        ),
+        entry.getSize,
+      )
+    }
+    val schema = new StructType()
+      .add(StructField("key", StringType))
+      .add(StructField("address", StringType))
+      .add(StructField("e_tag", StringType))
```

Comment:
```suggestion
      .add(StructField("etag", StringType))
```
(it's a word, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag)